### PR TITLE
Support fixing input qubits in encoder circuit synthesis

### PR DESCRIPTION
**Pull request by MaxieHelenBichmann**
## Description

The synthesis of encoding circuits for `CSSCode` and `StabilizerCode` was extended to enable fixing one or multiple input qubits to either $|0\rangle$ or $|+\rangle$, effectively synthesizing a circuit for a code with a lower $k$.

_Note 1_: As the circuit effectively encodes a different code than the given input, functions like `logical_to_input_mapping` do not have the expected behavior in that case.
_Note 2_: As there might be a (unrelated) bug in `StabilizerCode.__init__` in the case of no logical operators in the form of empty lists, the case that $k$ input qubits are fixed is not tested yet. 

Assisted-by: GitHub Copilot through inline suggestions; Codex via GPT-5.5 as first reviewer (both reviewed by me)

## Checklist

- [x] The pull request only contains commits that are focused and relevant to this change.
- [x] I have added appropriate tests that cover the new/changed functionality.
- [x] I have updated the documentation to reflect these changes.
- [x] The changes follow the project's style guidelines and introduce no new warnings.
- [x] The changes are fully tested and pass the CI checks.
- [x] I have reviewed my own code changes.

**If PR contains AI-assisted content:**

- [x] I have disclosed the use of AI tools in the PR description as per our [AI Usage Guidelines](https://github.com/munich-quantum-toolkit/qecc/blob/main/docs/ai_usage.md).
- [ ] AI-assisted commits include an `Assisted-by: [Model Name] via [Tool Name]` footer.
- [x] I confirm that I have personally reviewed and understood all AI-generated content, and accept full responsibility for it.

### CODE REVIEW

File: src/mqt/qecc/circuit_synthesis/encoding.py [hunk 1]
```diff
@@ -829,22 +829,41 @@ def synthesize_encoding_circuit(
     code: StabilizerCode,
     config: SynthesisConfig | None = None,
     use_cnots_if_css: bool = True,
+    fixed_logical_qubits: list[tuple[int, bool]] | list[int] | None = None,
```
Comment (pehamTom) [inline src/mqt/qecc/circuit_synthesis/encoding.py]: I think it would be better to have a more descriptive parameter here than a Bool. Maybe a str that can be either "X" (|+>) or "Z" (|0>). Or simply "0" and "+" (your choice).
This of course then requires checking that the string is valid. Going further, one might consider introducing a specific type for these kinds of situations as this appears in multiple places in QECC. But this is not something to be handled in this PR. 

I also think that since this is a somewhat niche feature, it is fine to demand the user to be explicit. Hence, I would drop the `list[int]` version. 

Nit: Also, instead of a list of tuples, this can just be a `dict`, I think.

File: src/mqt/qecc/circuit_synthesis/encoding.py [hunk 2]
```diff
@@ -829,22 +829,41 @@ def synthesize_encoding_circuit(
...
         code: The stabilizer code to synthesize the encoding circuit for.
         config: Configuration options for the synthesis process.
         use_cnots_if_css: Whether to use CNOT-only synthesis if the code is CSS.
+        fixed_logical_qubits: List of logical qubit indices that should be fixed to a logical zero state in the circuit, or list of tuples (index, is_zero) where is_zero indicates whether the logical qubit is fixed to a logical zero or plus state.
 
     Returns:
         A CliffordIsometry that implements the encoding circuit for the given stabilizer code.
     """
+    complete_fixed_logical_qubits = [
+        (item, True) if isinstance(item, int) else item for item in (fixed_logical_qubits or [])
```
Comment (pehamTom) [inline src/mqt/qecc/circuit_synthesis/encoding.py]: Nit: the `isinstance` check only has to be done once. 

But this point is moot if the above comment is considered.

File: tests/circuit_synthesis/test_encoder_synthesis.py [hunk 3]
```diff
@@ -405,3 +405,50 @@ def test_local_minimum() -> None:
     )
     enc = synthesize_encoding_circuit(code, config=config)
     assert enc.get_code().is_equivalent(code)
+
+
+@pytest.mark.parametrize("fixed_logical_qubits", [None, [], [0], [(0, True)], [(0, False)], [(1, True), (0, False)]])
```
Comment (pehamTom) [inline tests/circuit_synthesis/test_encoder_synthesis.py]: To make sure we also catch edge-cases:

- Test non-contiguous blocks of logicals (i.e. [1,6,4]).
- Test invalid logical assignment (this should fail)

File: tests/circuit_synthesis/test_encoder_synthesis.py [hunk 4]
```diff
@@ -405,3 +405,50 @@ def test_local_minimum() -> None:
...
+    """Check that fixed logical qubits are removed from the encoder inputs."""
+    encoder_cnots = synthesize_encoding_circuit(
+        css_4_2_2_code,
+        fixed_logical_qubits=fixed_logical_qubits,
+        use_cnots_if_css=True,
+    )
+    encoder_no_cnots = synthesize_encoding_circuit(
+        css_4_2_2_code,
+        fixed_logical_qubits=fixed_logical_qubits,
+        use_cnots_if_css=False,
```
Comment (pehamTom) [inline tests/circuit_synthesis/test_encoder_synthesis.py]: Nit: handle setting of this parameter via another parameter in the test.

**Review by pehamTom — DISMISSED** [review]
Thanks for your work here. This already looks pretty good. I have some rather minor suggestions for your consideration.

File: src/mqt/qecc/circuit_synthesis/encoding.py (hunk 1 above)
Comment (MaxieHelenBichmann) [reply src/mqt/qecc/circuit_synthesis/encoding.py]: Very valid point, anticipated something along those lines. Thanks for the feedback!

**Comment by lsschmid** [comment]
@coderabbit review
